main/cmds.go

```diff
@@ -1,17 +1,17 @@
 package main
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
-	"strconv"
 
 	"github.com/Azure/azure-docker-extension/pkg/vmextension"
 	"github.com/Azure/custom-script-extension-linux/pkg/seqnum"
 	"github.com/go-kit/kit/log"
 	"github.com/pkg/errors"
 )
 
-type cmdFunc func(*log.Context, vmextension.HandlerEnvironment) error
+type cmdFunc func(ctx *log.Context, hEnv vmextension.HandlerEnvironment, seqNum int) error
```

Comment:
Got it, it actually appears like only the `enable` function needs:

- sequence number (to save it to a file to prevent running again)
- configFolder (to locate the .settings files)

from HandlerEnvironment.xml. I might consider creating a smaller type.